websocket/connection.go

```diff
@@ -154,50 +161,53 @@ func (c *ManagedConnection) connect() (err error) {
 	return err
 }
 
-// keepalive keeps the connection open and reads control messages.
-// All messages are discarded.
-func (c *ManagedConnection) keepalive() (err error) {
+// keepalive keeps the connection open.
+func (c *ManagedConnection) keepalive() error {
+	for {
+		if err := c.read(); err != nil {
```

Comment:
```
var err
for err = c.read(); err != nil; err=c.read() {
}
return err
```

Reply:
This reads super weird, I'd prefer to keep the explicit version implemented currently if you don't mind? The `c.read()` twice throws me off.